test/integration/framework/test_server.go

```diff
@@ -99,7 +99,12 @@ func StartTestServer(t *testing.T, stopCh <-chan struct{}, setup TestServerSetup
 	if err != nil {
 		t.Fatalf("create temp file failed: %v", err)
 	}
-	defer os.RemoveAll(saSigningKeyFile.Name())
+	defer func() {
+		saSigningKeyFile.Close()
```

Comment:
Same here as above - won't t.Fatal() abort before it gets here?  I feel like this needs to defer after EACH step?

Reply:
thanks, done for that.